File: it-agents/skills/code-architect/references/code-examples/csharp/Facade.cs
// Facade — a simplified interface over a complex subsystem.
using System;

namespace Patterns.Facade
{
    // Subsystem parts, each with its own focused responsibility.
    sealed class SubsystemA { public string Step() => "A.ready"; }
    sealed class SubsystemB { public string Step(string input) => $"B.process({input})"; }
    sealed class SubsystemC { public string Step(string input) => $"C.finalize({input})"; }

    // Facade orchestrates the subsystems behind one simple call.
    sealed class Facade
    {
        private readonly SubsystemA _a = new();
        private readonly SubsystemB _b = new();
        private readonly SubsystemC _c = new();

        public string DoWork()
        {
            string r1 = _a.Step();
            string r2 = _b.Step(r1);
            return _c.Step(r2);
        }
    }

    static class Demo
    {
        public static void Run()
        {
            var facade = new Facade();
            Console.WriteLine(facade.DoWork());
        }

        public static void Main() => Run();
    }
}

File: it-agents/skills/code-architect/references/code-examples/java/Facade.java
// Facade — a simplified interface over a complex subsystem.

class Encoder {
    String encode(String raw) { return "[enc:" + raw + "]"; }
}
class Packetizer {
    String packetize(String data) { return "{pkt:" + data + "}"; }
}
class Transmitter {
    String send(String packet) { return "sent " + packet; }
}

// Facade hides the multi-step subsystem behind one call.
class MediaFacade {
    private final Encoder encoder = new Encoder();
    private final Packetizer packetizer = new Packetizer();
    private final Transmitter transmitter = new Transmitter();

    String broadcast(String message) {
        String encoded = encoder.encode(message);
        String packet = packetizer.packetize(encoded);
        return transmitter.send(packet);
    }
}

public class Facade {
    public static void main(String[] args) {
        MediaFacade facade = new MediaFacade();
        System.out.println(facade.broadcast("ping"));
    }
}
